### PR TITLE
pkgs/top-level/aliases.nix: improve error messages

### PR DESCRIPTION
**Pull request by xaverdh**
Change error messages in `aliases.nix` to start with or at least mention the name of the
package being referenced. This avoids obscure error messages like
"error: Abandoned by upstream." when rebuilding your system.
Also do some trivial cosmetic changes while touching things.

###### Motivation for this change

See https://github.com/NixOS/nixpkgs/pull/101098 for an example of the issue.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -483,16 +483,16 @@ mapAliases ({
...
+  ruby_2_0_0 = throw "ruby_2_0_0 was deprecated in 2018-0213: use a newer version of ruby";
+  ruby_2_1_0 = throw "ruby_2_1_0 was deprecated in 2018-0213: use a newer version of ruby";
+  ruby_2_2_9 = throw "ruby_2_2_9 was deprecated in 2018-0213: use a newer version of ruby";
+  ruby_2_3_6 = throw "ruby_2_3_6 was deprecated in 2018-0213: use a newer version of ruby";
+  ruby_2_3 = throw "ruby_2_3 was deprecated in 2019-09-06: use a newer version of ruby";
+  ruby_2_4_3 = throw "ruby_2_4_3 was deprecated in 2018-0213: use a newer version of ruby";
+  ruby_2_4 = throw "ruby_2_4 was deprecated in 2019-12: use a newer version of ruby";
+  ruby_2_5_0 = throw "ruby_2_5_0 was deprecated in 2018-0213: use a newer version of ruby";
+  rubyPackages_2_4 = throw "rubyPackages_2_4 was deprecated in 2019-12: use a newer version of rubyPackages instead";
+  rubygems = throw "rubygems was deprecated in 2016-03-02: rubygems is now bundled with ruby";
```
Comment (jtojnar) [inline pkgs/top-level/aliases.nix:495]: nit: Except for `ruby_2_4` and `rubyPackages_2_4`, these are days so they should use `on` preposition. You might also want to add a dash between `0213`.

Comment (xaverdh) [reply pkgs/top-level/aliases.nix]: fixed, thanks!

**Comment by xaverdh** [comment]
(rebased)

**Comment by jtojnar** [comment]
Nice, thank you.

**Pull request merged**